lib/matplotlib/patches.py

```diff
@@ -1619,12 +1617,15 @@ def iter_circle_intersect_on_line(x0, y0, x1, y1):
                 else:
                     sign_dy = 1.0
                 sqrt_discrim = np.sqrt(discrim)
-                for sign in (1., -1.):
-                    x = (D * dy + sign * sign_dy * dx * sqrt_discrim) / dr2
-                    y = (-D * dx + sign * np.abs(dy) * sqrt_discrim) / dr2
-                    yield x, y
+                return np.array(
+                    [[D * dy + sign_dy * dx * sqrt_discrim / dr2,
+                      -D * dx + abs(dy) * sqrt_discrim / dr2],
+                     [D * dy - sign_dy * dx * sqrt_discrim / dr2,
+                      -D * dx - abs(dy) * sqrt_discrim / dr2]])
```

Comment:
You lost the parentheses here around some additions.

Reply:
oops, that's embarassing, fixed